src/utils/pgwire/src/pg_message.rs

```diff
@@ -40,11 +41,18 @@ pub struct FeQueryMessage {
 }
 
 impl FeQueryMessage {
-    pub fn get_sql(&self) -> &str {
-        // Why there is a \0..
-        match std::str::from_utf8(&self.sql_bytes[..]) {
-            Ok(v) => v.trim_end_matches('\0'),
-            Err(e) => panic!("Invalid UTF-8 sequence: {}", e),
+    pub fn get_sql(&self) -> Result<&str> {
+        match CStr::from_bytes_with_nul(&self.sql_bytes) {
+            Ok(cstr) => cstr.to_str().map_err(|err| {
+                Error::new(
+                    ErrorKind::InvalidInput,
+                    format!("Cannot transform cstr to str: {}", err),
```

Comment:
It seems this error is for invalid utf-8 sequence, while the outer one is for bytes with no `nul` at the end.

```rust
core::ffi::c_str::CStr
pub fn to_str(&self) -> Result<&str, str::Utf8Error>

Yields a &[str] slice if the CStr contains valid UTF-8.
If the contents of the CStr are valid UTF-8 data, this function will return the corresponding &[str] slice. Otherwise, it will return an error with details of where UTF-8 validation failed.
```

Reply:
Thanks for mention, I fix in this pr https://github.com/singularity-data/risingwave/pull/2685.